core/src/main/java/io/vertx/maven/MavenCoords.java

```diff
@@ -23,22 +23,42 @@
 
   private final String owner;
   private final String serviceName;
+  private final String extension;
+  private final String classifier;
   private final String version;
 
-  public MavenCoords(String owner, String serviceName, String version) {
+  public MavenCoords(String owner, String serviceName, String extension, String classifier, String version) {
     this.owner = owner;
     this.serviceName = serviceName;
+    this.extension = extension;
+    this.classifier = classifier;
     this.version = version;
   }
 
   public MavenCoords(String idString) {
     String[] split = idString.split(":");
-    if (split.length != 2 && split.length != 3) {
-      throw new IllegalArgumentException("Invalid service identifier: " + idString);
+    if (split.length != 2 && split.length != 3 && split.length != 4 && split.length != 5) {
```

Comment:
split.length is repeated many times in this method - would be cleaner to create a local var for this